polaris-react/stylelint/plugins/coverage/rules/global-disallowed-list/index.js

```diff
@@ -0,0 +1,61 @@
+const {ruleMessages, validateOptions, report} = require('stylelint').utils;
+
+const {isRegExp, isString, namespace} = require('../../utils');
+
+const ruleName = namespace('global-disallowed-list');
+
+const messages = ruleMessages(ruleName, {
+  rejected: (value) => `Unexpected disallowed value "${value}"`,
+});
+
+const rule = (primary) => {
+  return (postcssRoot, postcssResult) => {
+    const validOptions = validateOptions(postcssResult, ruleName, {
+      actual: primary,
+      possible: [isRegExp, isString],
+    });
+
+    if (!validOptions) {
+      return;
+    }
+
+    postcssRoot.walkDecls((decl) => {
+      let currentTest;
+      const hasDisallowed = primary.some((test) => {
+        currentTest = test;
+        return testStringOrRegEx(test, decl.value);
+      });
```

Comment:
Wonder if we should accumulate all disallowed matches in a declaration and report. Otherwise, folks might have to run Stylelint multiple times to resolve all issues. e.g.

```js
const allDisallowed = [];

primary.forEach((test) => {
  if (testStringOrRegEx(test, decl.value)) {
    allDisallowed.push({test, value: decl.value})
  }
});

if (!allDisallowed.length) return

report(...)
```

Reply:
Excellent suggestion! Done 👍